convention.md

```diff
@@ -18,11 +18,50 @@ The underscore (`_`) is used and reserved for Homie *node arrays*.
 
 ### Payload
 
-Every MQTT message payload MUST be sent as UTF-8 encoded string.
-A numeric data type MUST be converted to a literal string with the dot character as the decimal separator for the `float` data type.
-An empty message is not valid as payload for numeric data types.
-The value published as payload MUST be valid for the respective property/attribute type as per the convention definition below.
-Booleans MUST be converted to "true" or "false".
+#### General Specification
+ - Every MQTT message payload MUST be sent as a UTF-8 encoded string
+-  The value published as payload MUST be valid for the respective property/attribute type as per the defined convention
+ 
+ #### String
+ - String types are limited to 268,435,456 characters 
+ - An empty string ("") is a valid payload
+ 
+ #### Integer
+ - Integer types are UTF-8 encoded representations of 64-bit signed whole numbers
+ - Integers range from -9,223,372,036,854,775,808 (-2<sup>63</sup>) to 9,223,372,036,854,775,807 (2<sup>63</sup>-1)
```

Comment:
This looks like a signed 128bit number ?? Should it actually be a 64 bit signed number not 128 ?

Reply:
I am basing it off the numbers here: 
https://docs.oracle.com/javase/7/docs/api/constant-values.html#java.lang.Long.MAX_VALUE

Reply:
Dont know what i was thinking, your right that is a 64bit number lol. Think in arduino land a long is 32bit, and a long long is 64bit. We're okay with max 64bit number ?

Reply:
Not sure.  All of my definitions are up for discussion.  I created them based on my assumptions while reading the spec. I *think* a 64 bit number is ok.  Anything bigger than that gets complicated and may reduce targetable platforms.